### ISSUE TITLE
processDirectories() fails to return any directories if .gitignore contains directory/project/binary name

### ISSUE THREAD

**Issue by haukened** [wailsapp/wails#2867]
### Description

If you are building on linux with Go, go will produce a binary of the same name as the project top level directory.
Adding the name of this binary to the `.gitignore` file will cause `processDirectories()` to treat this as a wild card under the top level folder, effectively ignoring filesystem change notifications for the entire project.

### To Reproduce

(assumes project named `foo` under `/home/you/foo`)
1. Create wails project `cd && wails init -n foo -t <whichever>`
2. add binary name `foo` to .gitignore
3. run `wails dev`
4. observe the stunning silence from `*fsnotify.Watcher` as you frantically edit every go file trying to figure out what is happening.

### Expected behaviour

the `processDirectories()` function applies only to sub-directories of the top level folder, matching the behavior or `git` for which the `.gitignore` file is designed.

### System Details

```shell
Wails CLI v2.5.1

 SUCCESS  Done.                                                                                                                                             

# System

OS           | Arch Linux
Version      | Unknown   
ID           | arch      
Go Version   | go1.21.0  
Platform     | linux     
Architecture | amd64     

# Wails

Version         | v2.5.1
Package Manager | pacman

# Dependencies

Dependency | Package Name | Status    | Version    
*docker    | docker       | Installed | 1:24.0.5-1 
gcc        | gcc          | Installed | 13.2.1-3   
libgtk-3   | gtk3         | Installed | 1:3.24.38-1
libwebkit  | webkit2gtk   | Installed | 2.40.5-1   
npm        | npm          | Installed | 9.8.1-1    
pkg-config | pkgconf      | Installed | 1.8.1-1    
* - Optional Dependency

# Diagnosis

Your system is ready for Wails development!
```

**Comment by haukened** [comment]
@leaanthony you can assign this one back to me, i just wanted it to be documented. Will tackle tomorrow.

**Comment by leaanthony** [comment]
There was a recent PR around .gitignore files...

**Comment by haukened** [comment]
This was pretty simple to solve, just by grabbing the base level directory with `filepath.Base(cwd)` and then putting an `if` on the `ignoreDirs` append to filter the append of the base directory (and consequently the binary name).

**Comment by haukened** [comment]
Resolved in [PR #2869](https://github.com/wailsapp/wails/pull/2869)
